app/weather_forecast.py

```diff
@@ -0,0 +1,265 @@
+
+
+def validate_date_input(day, month, year):
+    """ date validation.
+    Args:
+        day (int / str) - day part of date.
+        month (int / str) - month part of date.
+        year (int / str) - year part of date.
+    Returns:
+        (bool) - validate ended in success or not.
+        day (int) - day part of date.
+        month (int) - month part of date.
+        year (int) - year part of date.
+    """
+    try:
+        day = int(day)
+        month = int(month)
+        year = int(year)
+    except ValueError:
+        return False, day, month, year
+    if 1975 <= year <= 2050:
+        try:
+            datetime(year=year, month=month, day=day)
+        except ValueError:
+            return False, day, month, year
+    else:
+        return False, day, month, year
+    return True, day, month, year
+
+
+def get_data_from_api(url, input_query_string):
+    """ get the relevant weather data by calling the "Visual Crossing Weather" API.
+    Args:
+        url (str) - API url.
+        input_query_string (dict) - input for the API.
+    Returns:
+        success_in_get_weather_data (bool) - did the API call ended in success or failure (location not found etc).
+        response_json (json dict) - relevant part (data / error) of the JSON returned by the API.
+    """
+    load_dotenv()
+    HEADERS['x-rapidapi-key'] = getenv('WEATHER_API_KEY')
+    success_in_get_weather_data = True
+    response = requests.request("GET", url, headers=HEADERS, params=input_query_string)
+    try:
+        response_json = response.json()["locations"]
+    except KeyError:
+        success_in_get_weather_data = False
+        response_json = response.json()
+    return success_in_get_weather_data, response_json
+
+
+def get_historical_weather(input_date, location):
+    """ get the relevant weather from history by calling the API.
+    Args:
+        input_date (date) - day part of date.
+        location (str) - location name.
+    Returns:
+        (int) - minimum degrees in Celsius.
+        (int) - maximum degrees in Celsius.
+        (str) - weather conditions.
+        (str) - location / error description.
+    """
+    input_query_string = BASE_QUERY_STRING
+    input_query_string["startDateTime"] = input_date.isoformat()
+    input_query_string["endDateTime"] = (input_date + timedelta(days=1)).isoformat()
+    input_query_string["location"] = location
+    success_in_get_weather_data, api_json = get_data_from_api(HISTORY_URL, input_query_string)
+    if success_in_get_weather_data:
+        for item in api_json:
+            # print("historical:", api_json[item]['values'][0]['mint'], api_json[item]['values'][0]['maxt'])
+            min_temp = api_json[item]['values'][0]['mint']
+            max_temp = api_json[item]['values'][0]['maxt']
+            conditions = api_json[item]['values'][0]['conditions']
+        return min_temp, max_temp, conditions, api_json[item]['address']
```

Comment:
From the indent of the return, the values of only the last instance are returned. Is that the intention? If so, then there isn't a need to iterate over the whole list.

Also remove commented out code.